test/Microsoft.Sbom.Targets.E2E.Tests/GenerateSbomE2ETests.cs

```diff
@@ -0,0 +1,344 @@
+        var sampleProject = new Project(projectFile);
+
+        // Get all the expected default properties
+        SetDefaultProperties(sampleProject);
+
+        // Set the TargetFrameworks property to empty. By default, it sets this property to net6.0 and net8.0, which fails for net8.0 builds.
+        sampleProject.SetProperty("TargetFrameworks", string.Empty);
+
+        // Set the paths to the sbom-tool CLI tool and Microsoft.Sbom.Targets.dll
+        sampleProject.SetProperty("SbomToolBinaryOutputPath", sbomToolPath);
+        sampleProject.SetProperty("GenerateSbomTaskAssemblyFilePath", generateSbomTaskPath);
+
+        return sampleProject;
+    }
+
+    private void SetDefaultProperties(Project sampleProject)
+    {
+        expectedPackageName = sampleProject.GetPropertyValue("PackageId");
+        expectedVersion = sampleProject.GetPropertyValue("Version");
+        assemblyName = sampleProject.GetPropertyValue("AssemblyName");
+        configuration = sampleProject.GetPropertyValue("Configuration");
+
+        if (string.IsNullOrEmpty(expectedPackageName))
+        {
+            expectedPackageName = assemblyName;
+        }
+
+        if (string.IsNullOrEmpty(expectedPackageName))
+        {
+            expectedVersion = "1.0.0";
+        }
+    }
+
+    private void RestoreBuildPack(Project sampleProject)
+    {
+        var logger = new ConsoleLogger();
+
+        // Restore the project to create project.assets.json file
+        var restore = sampleProject.Build("Restore", new[] { logger });
+        Assert.IsTrue(restore, "Failed to restore the project");
+
+        // Next, build the project
+        var build = sampleProject.Build(logger);
+        Assert.IsTrue(build, "Failed to build the project");
+
+        // Finally, pack the project
+        var pack = sampleProject.Build("Pack", new[] { logger });
+        Assert.IsTrue(pack, "Failed to pack the project");
+    }
+
+    private void ExtractPackage()
+    {
+        // Unzip the contents of the NuGet package
+        var nupkgPath = Path.Combine(projectDirectory, "bin", configuration);
+        var nupkgFile = Path.Combine(nupkgPath, $"{expectedPackageName}.{expectedVersion}.nupkg");
+        var zipFile = Path.Combine(nupkgPath, $"{expectedPackageName}.{expectedVersion}.zip");
+        var extractPath = Path.Combine(projectDirectory, "bin", configuration, $"{Guid.NewGuid()}.temp");
+
+        // Rename the .nupkg file to .zip
+        File.Copy(nupkgFile, zipFile, true);
+
+        // Extract the .zip file
+        ZipFile.ExtractToDirectory(zipFile, extractPath);
+
+        manifestPath = Path.Combine(extractPath, "_manifest", sbomSpecificationDirectoryName, "manifest.spdx.json");
+    }
+
+    [TestMethod]
+    public void SbomGenerationSucceedsForDefaultProperties()
+    {
+        if (!IsWindows)
+        {
+            Assert.Inconclusive("This test is not (yet) supported on non-Windows platforms.");
```

Comment:
These tests kept failing for MacOS because it could not find the "MonoPosixHelper" dependency when invoking our task: https://github.com/microsoft/sbom-tool/actions/runs/10381644120/job/28743400968 

Unfortunately, we were not able to root cause this issue, so we made these tests only run for Windows for now